en/services/mission.md

```diff
@@ -297,10 +302,57 @@ The protocol has been implemented in C.
 Source code:
 * [src/MissionManager/PlanManager.cc](https://github.com/mavlink/qgroundcontrol/blob/master/src/MissionManager/PlanManager.cc)
 
+
 ### ArduPilot
 
+ArduPilot implements the mission protocol in C.
+While the messages and sequence are the same, there are a number of significant differences from  the behaviour described above.
+
+Source:
+* [/libraries/GCS_MAVLink/GCS_Common.cpp](https://github.com/ArduPilot/ardupilot/blob/master/libraries/GCS_MAVLink/GCS_Common.cpp)
+
+
+#### Flight Plan Missions
+
+Mission upload, download, clearing missions, and monitoring progress and partial mission upload ([MISSION_WRITE_PARTIAL_LIST](#MISSION_WRITE_PARTIAL_LIST)) are supported.
+
+Partial mission download is not supported ([MISSION_REQUEST_PARTIAL_LIST](#MISSION_REQUEST_PARTIAL_LIST)).
+
+ArduPilot's implementation differs from this specification (non-exhaustively):
+- The mission sequence number (`seq` in commands) of 0 is populated with the home position of the vehicle.
+  - This is an invalid sequence index with respect to the specification, and can be dropped by compliant systems.
+  - ArduPilot will silently drop a mission item with `seq==0` and an "invalid" home position.
+  - Index 1 and later are the mission items, as per the specification.
+
+
+- Mission uploads are not "robust" (i.e. after a mission upload the stored mission may not match the uploaded mission). For example:
+  - If you try and upload more items than ArduPilot can store the system will "accept" the items (i.e. not report a failure) but will just overwrite each new item to the same (highest) slot in the mission list.
+  
+In addition, the following behaviour is not defined by the specification:
+- ArduPilot performs some validation of fields when mission items are submitted. 
+  The validation code is common to all vehicles; mission items that are not understood by the vehicle type are accepted on upload but skipped during mission execution.
+- A new mission can be uploaded while a mission is being executed. 
```

Comment:
Note to me: Need to check if this is common behaviour. Might be good to specify what should happen here.